src/vs/workbench/contrib/terminal/browser/media/shellIntegration.zsh

```diff
@@ -2,9 +2,23 @@
 #   Copyright (c) Microsoft Corporation. All rights reserved.
 #   Licensed under the MIT License. See License.txt in the project root for license information.
 # ---------------------------------------------------------------------------------------------
-
 autoload -Uz add-zsh-hook
 
+. ~/.zshenv
+if [ -z "$VSCODE_SHELL_LOGIN" ]; then
+	if [ -f ~/.zprofile ]; then
+		. ~/.zprofile
+	fi
+fi
+. ~/.zshrc
+if [ -z "$VSCODE_SHELL_LOGIN" ]; then
+	if [ -f ~/.zlogin ]; then
+		. ~/.zlogin
+		VSCODE_SHELL_LOGIN=""
```

Comment:
I think we want `unset VSCODE_SHELL_LOGIN` here? Also there should be another `if [ -z "$VSCODE_SHELL_LOGIN" ]` block at the end since the variable won't be cleared if `~/.zlogin` doesn't exist currently (would also help a bit with readability imo).

Reply:
I think it's safe to always unset both `VSCODE_SHELL_LOGIN` and `ZDOTDIR` there.